app/static/js/highlight-nav-links.js

```diff
@@ -0,0 +1,44 @@
+const sections = document.querySelectorAll("section[id]")
+const navElements = document.querySelectorAll(".nav-element")
+let inViewId = ""
+
+const handleIntersection = (entries) => {
+    entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+            const sectionId = entry.target.getAttribute("id")
+            inViewId = sectionId
+            updateNavigationView(inViewId)
+        }
+    })
+}
+
+const navPositionObserverOptions = {
+    root: null,
+    rootMargin: "0px",
+    threshold: 0.2,
+}
+
+const navPositionObserver = new IntersectionObserver(
+    handleIntersection,
+    navPositionObserverOptions,
+)
+
+sections.forEach((section) => {
+    navPositionObserver.observe(section)
+})
+
+const updateNavigationView = (inViewId) => {
```

Comment:
"Erhebung 2" is never really highlighted for me – the highlighting jumps from "Erhebung 1" straight to "Erhebung 3". But it could be that this will already work better once we have more content (right now, each section is quite short anyway).